File: languages/nim/worker.nim
import strutils
const key=0xDF
proc transform(h:string):string =
  result=""; var i=0
  while i<h.len:
    let v=parseHexInt(h[i..i+1]) xor key
    result.add(toHex(v,2).toLowerAscii); i+=2
while true:
  try:
    let line=stdin.readLine.strip(leading=false,trailing=true,chars={'\r','\n'})
    if line=="PING": echo "PONG"; stdout.flushFile(); continue
    if line=="QUIT": break
    if line.len>=2 and (line[0]=='E' or line[0]=='D') and line[1]==' ': echo transform(line[2..^1]); stdout.flushFile()
    else: echo "ERR"; stdout.flushFile()
  except EOFError: break

File: languages/kotlin/Worker.kt
import java.io.BufferedReader
import java.io.InputStreamReader
fun main(){val key=0x31;val r=BufferedReader(InputStreamReader(System.`in`));while(true){val s=r.readLine()?:break;if(s=="PING"){println("PONG");System.out.flush();continue};if(s=="QUIT")break;if(s.matches(Regex("^[ED] [0-9A-Fa-f]*$"))){val h=s.substring(2);val o=StringBuilder();var i=0;while(i<h.length){o.append(((h.substring(i,i+2).toInt(16) xor key) and 255).toString(16).padStart(2,'0'));i+=2};println(o.toString());System.out.flush()}else{println("ERR");System.out.flush()}}}
